stuff2sort/System.py

```diff
@@ -0,0 +1,1150 @@
+            self.charge_vector.extend(self.readcharges(element))
+
+        #   Read All Atom Lists
+        #   XX AJ currently I'm assuming we're always having separate files for atom indices, only remove this comment when we finally decided so or add the possibility for lists
+        self.qmatomslist = self.read_atoms_list(self.input_dict['qmatomslist'])
+        self.activeatomslist = self.read_atoms_list(self.input_dict['activeatomslist'])
+        if input_dict['useinnerouter']:
+            self.inneratomslist = self.read_atoms_list(self.input_dict['inneratomslist'])
+            self.outeratomslist = self.read_atoms_list(self.input_dict['outeratomslist'])
+        #   If we're not using inner/outer we're keeping the lists empty (currently necessary for some functions but that might change later)
+        else:
+            self.inneratomslist = []
+            self.outeratomslist = []
+
+        #   Read Connectivity
+        self.connectivity_list = self.read_connectity_from_topology(self.input_dict['topologyfile'])
+
+        #   Read Initial Geometry
+        #   XX AJ I would prefer only one function here independent of the file type and only make that distinction within the function. I'll get back to that later when I'm writing GeneratorGeometries.py
+        if self.input_dict['coordinatefile'][-4:] == ".gro":
+            #logger(logfile, "Reading geometry (.gro)...\n")
+            # XX AJ if we rewrite gro files to g96 files, the following function can be deleted and we can read the g96 file with geometry.readg96 afterwards
+            self.initial_geometry = geometry.readgeo(self.input_dict['coordinatefile'])
+            # Writing high-precision coordinate file
+            # logger(logfile, "Writing high-precision coordinate file...")
+            self.write_highprec(self.input_dict['coordinatefile']) # XX temp removed logfile until logfile decision of Florian AJ
+            self.input_dict['coordinatefile'] = self.input_dict['jobname'] + ".g96"
+            # logger(logfile, "Done.\n")
+        elif self.input_dict['coordinatefile'][-4:] == ".g96":
+            #logger(logfile, "Reading geometry (.g96)...\n")
+            self.initial_geometry = geometry.readg96(self.input_dict['coordinatefile'])
+
+        self.number_of_atoms = int(len(self.initial_geometry)/3)
+
+        #   Create xyzq (Coordinates And Charges) For The Whole System
+        #   XX AJ same here, I'll make one function of it
+        #   XX AJ technically, I would prefer this xyzq function not in this class, but it's used in the get_linkatoms_ang, so I'll keep it here
+        if input_dict['useinnerouter']:
+            self.xyzq = geometry.make_xyzq_io(self.initial_geometry, self.charge_vector, self.outeratomslist)
+        else:
+            self.xyzq = geometry.make_xyzq(self.initial_geometry, self.charge_vector)
+
+        #   Read linkatoms and next order atoms
+        self.m1list = self.identify_m1()
```

Comment:
self.list_atoms_m1
self.list_atoms_m2
 
I propose we use different names for m1 and m2; it is not directly obvious what m1 and m2 are.

Reply:
I'm not sure, I think for people actually looking into the code it's enough to explain m1/m2 in the documentation. Also there is the comment above to describe the lists. I just can't think of a good variable name that describes m1/m2 atoms withour getting too long.

Reply:
Add a comment in the code that more explanation is in the documenation

Reply:
Done